### PR TITLE
Use final members for thread visibility. And close the InputStream.

### PR DESCRIPTION
**Pull request by baldersheim**
@ollivir or @geirst PR
@jobergum FYI

### CODE REVIEW

File: fsa/src/main/java/com/yahoo/fsa/FSA.java [hunk 1]
```diff
@@ -339,50 +340,49 @@ public Iterator iterator(State state) {
...
-        if ( ! "file".equals(fsaUrl.getProtocol()))
+        if ( ! "file".equals(fsaUrl.getProtocol())) {
             throw new RuntimeException("Could not open non-file url '" + fsaUrl + "' as a file input stream: " +
-                                       "The classloader of " + loadingClass + "' does not return file urls");
+                    "The classloader of " + loadingClass + "' does not return file urls");
+        }
         return new FSA(fsaUrl.getFile());
     }
 
+    static FileInputStream createInputStream(String filename) {
```
Comment (ollivir) [inline fsa/src/main/java/com/yahoo/fsa/FSA.java]: `private` maybe?

**Review by ollivir — APPROVED** [review]
One comment on `createInputStream`, but otherwise looks good to me.

File: fsa/src/main/java/com/yahoo/fsa/FSA.java (hunk 1 above)
Comment (baldersheim) [reply fsa/src/main/java/com/yahoo/fsa/FSA.java]: Done

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**